client/src/store/index.js

```diff
@@ -33,10 +34,49 @@ export const actions = {
       dispatch('getDailyData');
     });
   },
+  getSelectedLanguagesWithFlags({ commit, dispatch }) {
+    commit('setLoading', true);
+    axios
+      .get(`${process.env.VUE_APP_API_URL}/languages`)
+      .then(result => {
+        chrome.storage.sync.get('selectedLanguages', response => {
+          const { selectedLanguages } = response;
+          let selectedLanguagesWithFlags = [];
+          if (
+            selectedLanguages === undefined ||
+            selectedLanguages.length === 0
+          ) {
+            selectedLanguagesWithFlags = [
+              {
+                language: 'german',
+                flag: 'twa-germany-flag',
+              },
+            ];
+          } else {
+            selectedLanguagesWithFlags = selectedLanguages.map(o => {
+              return {
+                language: o,
+                flag: result.data.languages[o].flag,
+              };
+            });
+          }
+          dispatch(
+            'saveSelectedLanguagesWithFlags',
+            selectedLanguagesWithFlags,
+          );
+          dispatch('getDailyData');
+        });
+      })
+      .catch(() => commit('setHasError', true));
+  },
   saveSelectedLanguages({ commit }, selectedLanguages) {
     chrome.storage.sync.set({ selectedLanguages });
     commit('setSelectedLanguages', selectedLanguages);
   },
+  saveSelectedLanguagesWithFlags({ commit }, selectedLanguagesWithFlags) {
+    chrome.storage.sync.set({ selectedLanguagesWithFlags });
```

Comment:
What do you think about making `selectedLanguages` an array of objects instead? This is so we won't need to have two separate keys that store the same thing.